pkg/wf/extractor.go

```diff
@@ -380,8 +433,64 @@ func (extractor *Extractor) extract() error {
 	return extractor.writePathList(pathList)
 }
 
+func (extractor *Extractor) pack() error {
+	err := os.MkdirAll(extractor.config.DestPath, 0777)
+	if err != nil {
+		return err
+	}
+
+	paths, err := extractor.getInitialPaths()
+	if err != nil {
+		return err
+	}
+	items := []*concurrency.Item[*extractParams, [][]byte]{{Output: paths}}
+	seenPaths := map[string]bool{}
+
+	err = concurrency.Dispatcher(
+		func(i *concurrency.Item[*extractParams, [][]byte]) ([]*concurrency.Item[*extractParams, [][]byte], error) {
+			var output []*concurrency.Item[*extractParams, [][]byte]
+			if i.Output != nil {
+				for _, p := range i.Output {
+					if !seenPaths[string(p)] {
+						seenPaths[string(p)] = true
+						output = append(output, &concurrency.Item[*extractParams, [][]byte]{
+							Data: &extractParams{
+								path:    string(p),
+								parsers: extractor.parsers,
+								config:  extractor.config,
+							},
+							Output: nil,
+							Err:    nil,
+						})
+					}
+				}
+			}
+			return output, nil
+		},
```

Comment:
This method of path calculation is likely inefficient for packing because we still construct extraction paths but instead write files in reverse. However I do not have a very clear idea at the moment of how to make reverse path calculation convenient for any directory structure that isn't a result of extracting with default configurations (or even allowing manually specifying target paths). 🤔